google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITE2ETracingTest.java

```diff
@@ -0,0 +1,244 @@
+import static org.junit.Assert.assertEquals;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOpenTelemetryOptions;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.opentelemetry.trace.TraceConfiguration;
+import com.google.cloud.opentelemetry.trace.TraceExporter;
+import com.google.cloud.trace.v1.TraceServiceClient;
+import com.google.common.base.Preconditions;
+import com.google.devtools.cloudtrace.v1.Trace;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ITE2ETracingTest {
+
+  private static final Logger logger = Logger.getLogger(ITBaseTest.class.getName());
+
+  // TODO(jimit) Generalize project ID, accept from config
+  private static final String PROJECT_ID = "jimit-test";
+
+  private static final int NUM_TRACE_ID_BYTES = 32;
+
+  private static final int NUM_SPAN_ID_BYTES = 16;
+
+  private static final int GET_TRACE_RETRY_COUNT = 5;
+
+  private static final int GET_TRACE_RETRY_BACKOFF_MILLIS = 1000;
+
+  private static final int TRACE_PROVIDER_SHUTDOWN_MILLIS = 1000;
+
+  // Random int generator for trace ID and span ID
+  private static Random random;
+
+  private static TraceExporter traceExporter;
+
+  // Required for reading back traces from Cloud Trace for validation
+  private static TraceServiceClient traceClient_v1;
+
+  private static String rootSpanName;
+  private static Tracer tracer;
+
+  // Required to set custom-root span
+  private static OpenTelemetrySdk openTelemetrySdk;
+
+  private static Firestore firestore;
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    // Set up OTel SDK
+    Resource resource =
+        Resource.getDefault().merge(Resource.builder().put(SERVICE_NAME, "Sparky").build());
+
+    // TODO(jimit) Make it re-usable w/ InMemorySpanExporter
+    traceExporter =
+        TraceExporter.createWithConfiguration(
+            TraceConfiguration.builder().setProjectId(PROJECT_ID).build());
```

Comment:
I believe you should be able to remove `PROJECT_ID` usages in this file. You can use `firestoreOptions.getProjectId()` so that the two services (Firestore and CloudTrace) are used in the same gcp project. I think the CI picks that up from env vars [here](https://github.com/googleapis/java-firestore/blob/434a55b478fc617c2ec892a30d3eb1ce027ac54f/.kokoro/presubmit/integration.cfg) and [here](https://github.com/googleapis/java-firestore/blob/434a55b478fc617c2ec892a30d3eb1ce027ac54f/.kokoro/presubmit/integration-named-db.cfg)

Reply:
Do you mean `FirestoreOptions.getDefaultProjectId()`?